contracts/vaults/RibbonDeltaVault.sol

```diff
@@ -146,7 +147,36 @@ contract RibbonDeltaVault is RibbonVault, OptionsDeltaVaultStorage {
      * @notice Closes the existing long.
      *         This allows all the users to withdraw if the next option is malicious.
      */
-    function commitAndClose() external onlyOwner nonReentrant {}
+    function commitAndClose() external onlyOwner nonReentrant {
+        address oldOption = optionState.currentOption;
+
+        optionState.nextOption = counterpartyThetaVault
+            .optionState()
+            .nextOption;
+        optionState.nextOptionReadyAt = uint32(block.timestamp.add(delay));
```

Comment:
We could streamline the state updates by stuffing these into `_closeLong` instead